src/main/java/com/example/demo/db/entities/AvailableShifts.java

```diff
@@ -1,15 +1,19 @@
 package com.example.demo.db.entities;
 
+import com.example.demo.db.entities.interfaces.IEntity;
+import com.example.demo.db.entities.interfaces.Validatable;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
 import jakarta.validation.constraints.NotNull;
 
 import java.io.Serializable;
+import java.util.Collection;
+import java.util.Objects;
 
 @Entity
-public class AvailableShifts implements Serializable {
+public class AvailableShifts implements Serializable , Comparable<AvailableShifts>, IEntity<AvailableShifts,Integer> , Validatable<AvailableShifts> {
```

Comment:
Redundant Implement field they are all inside `IEntity`
```suggestion
public class AvailableShifts implements IEntity<AvailableShifts,Integer>  {
```